query/query.go

```diff
@@ -129,6 +130,11 @@ func (r *results) Next() <-chan Result {
 	return r.res
 }
 
+func (r *results) NextSync() (Result, bool) {
+	val, ok := <-r.res
+	return val, ok
```

Comment:
Never mind, I was going off the comments you gave me in IRC.  If we require the user calls close all is well.